Clarify guidelines around use of role pings in discord (#67639)
tgstation/tgstation: .github/CONTRIBUTING.md

Note: VSC requires an [extension](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) to take advantage of editorconfig.

Github actions that require additional configuration are disabled on the repository until ACTION_ENABLER secret is created with non-empty value.

> ## Using the maintainer role ping in discord

Reviewer: ```suggestion
## Using the Maintainer Role Ping in Discord
```

> This role is pingable as a compromise reached with the server host mso over the auto staling system. It should be used only to ping maintainers when your PR has had the stale label applied. Using it before then can be met with escalating timeouts and referall to the discord moderators for further infractions.

Reviewer: ```suggestion
This role `@Maintainer` is pingable as a compromise reached with the server host MrStonedOne over the auto-staling system we presently have in the codebase. It should be used only to ping Maintainers when your PR has had the "Stale" label applied. Using it before then can be met with escalating timeouts and referral to /tg/station's Discord moderators for further infractions.
```

> This role `@Maintainer` is pingable as a compromise reached with the server host MrStonedOne over the auto-staling system we presently have in the codebase. It should be used only to ping Maintainers when your PR has had the "Stale" label applied. Using it before then can be met with escalating timeouts and referral to /tg/station's Discord moderators for further infractions.

Reviewer 2: ```suggestion
This role `@Maintainer` is pingable as a compromise reached with the server host MrStonedOne over the auto-stale system we presently have in the codebase. It should be used only to ping Maintainers when your PR has had the "Stale" label applied. Using it before then can be met with escalating timeouts and referral to /tg/station's Discord moderators for further infractions.
```

staling is more grammatically correct, but even after a nights sleep it kept reading it as scaling or stalling.

maybe my brain just hates the word stale.

> Feel free to engage in general in coding general without the role ping, before your PR goes stale to build interest and get reviews.

Reviewer: ```suggestion
Feel free to engage and obtain general feedback in the Coding General channel without the role ping before your PR goes stale to build interest and get reviews.
```
